plugins/inputs/etherscan/etherscan.go

```diff
@@ -0,0 +1,244 @@
+package etherscan
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/models"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type EtherscanOption func(*Etherscan)
+
+func WithHTTPClient(newClient HTTPClient) EtherscanOption {
+	return func(plugin *Etherscan) {
+		plugin.client = newClient
+	}
+}
+
+func WithLogger(newLogger telegraf.Logger) EtherscanOption {
+	return func(plugin *Etherscan) {
+		plugin.log = newLogger
+	}
+}
+
+type Etherscan struct {
+	log telegraf.Logger
+
+	client          HTTPClient
+	Timeout         string `toml:"req_timeout_duration"`
+	timeoutDuration time.Duration
+
+	networkRequests requestQueue
+	Key             string `toml:"api_key" required:"true"`
+	requestCount    int
+
+	RequestLimit            int    `toml:"api_rate_limit_calls"`
+	RequestInterval         string `toml:"api_rate_limit_duration"`
+	requestIntervalDuration time.Duration
```

Comment:
Again
```suggestion
	RequestInterval         internal.Duration `toml:"api_rate_limit_duration"`
```
so that telegraf does the parsing for you.